### ISSUE TITLE
addMany templateFiles Array 

### ISSUE THREAD

**Issue by MarcZenn** [amwmedia/plop#83]
Hi 

I'm very new to plopjs and am having trouble with the `addMany` built-in action. 

Currently I am working on a monorepository which contains a boilerplate react application. 

I want to copy that entire directory (which contains a boilerplate react application) and place a copy of it elsewhere inside the same monorepo. 

Here is my current plopfile.js: 

```
module.exports = ( plop ) => {
  const templateFiles = [
    "packages/react-redux-boilerplate-package/*.js",
    "packages/react-redux-boilerplate-package/*.jsx",
    "packages/react-redux-boilerplate-package/*.json",
    "packages/react-redux-boilerplate-package/*.md",
    "packages/react-redux-boilerplate-package/.gitignore",
    "packages/react-redux-boilerplate-package/public/*",
    "packages/react-redux-boilerplate-package/src/components/*.js",
    "packages/react-redux-boilerplate-package/src/containers/app/*",
    "packages/react-redux-boilerplate-package/src/containers/*.js",
    "packages/react-redux-boilerplate-package/src/redux/actions/*.js",
    "packages/react-redux-boilerplate-package/src/redux/reducers/*.js",
    "packages/react-redux-boilerplate-package/src/redux/sagas/*.js",
    "packages/react-redux-boilerplate-package/src/redux/*.js",
    "packages/react-redux-boilerplate-package/src/utils/*.js"
  ]

  // Declare a new generator called "newApp" for use with our react-redux-boilerplate app
  plop.setGenerator( "newApp", {

    // Succintly describes what generator does.
    description: "Create a new lerna package aka new app directory",

    // Get inputs from the user.
    // That's Inquirer.js doing the job behind the hood.
    prompts: [
      {
        type: "input",
        name: "name",
        message: "What is your app name?"
      }
    ],

    // List of actions to take.
    // Here we "add" new files from our templates or boilerplates.
    actions: [
      {
        type: "addMany",
        destination: "packages/{{lowerCase name}}",
        base: 'packages/react-redux-boilerplate-package',
        templateFiles: templateFiles
      },
    ]

  } );
};
```
and here is my the boilerplate application file structure: 

```
.
├── README.md
├── package.json
├── public
│   └── index.html
├── src
│   ├── components
│   │   └── index.js
│   ├── containers
│   │   ├── app
│   │   │   ├── app.container.jsx
│   │   │   ├── app.container.scss
│   │   │   └── app.container.test.jsx
│   │   └── index.js
│   ├── index.js
│   ├── redux
│   │   ├── actions
│   │   │   ├── app.actions.js
│   │   │   └── index.js
│   │   ├── index.js
│   │   ├── reducers
│   │   │   ├── app.reducer.js
│   │   │   └── index.js
│   │   └── sagas
│   │       └── index.js
│   ├── router.jsx
│   ├── routes.js
│   └── utils
│       └── index.js
└── webpack.config.js
```
When I run plop and go through the command line prompt I get the following error:

`[FAILED] addMany patterns must be a string or an array of strings`

I understand that the `addMany` actions is relatively new so perhaps I'm missing something and using it incorrectly or there simply isn't support for complex directories yet? If my question is better suited for StackOverflow I'm happy to close this out and post it there.

**Comment by amwmedia** [comment]
Thanks for all the detail! I'll look into this. @nicoespeon any initial thoughts?

**Comment by amwmedia** [comment]
I found the issue. Currently, the `addMany` action only accepts a string instead of an array of strings because when we [pass the temlpateFiles to globby](https://github.com/amwmedia/node-plop/blob/master/src/actions/addMany.js#L30), we're putting it into an array. We'll get a fix in soon. Sorry for the trouble, hopefully now that you understand the issue, you'll be able to work around it some.

**Comment by nicoespeon** [comment]
Thanks @amwmedia, you solved this quicker than me having a look at the question 😄

**Comment by amwmedia** [comment]
😉 turned out to be pretty simple, but at first glance, I wasn't sure.

**Comment by MarcZenn** [comment]
Wow thank you so much guys for the quick response! This definitely helps. 

I'll keep an eye out for the fix. 

Cheers!

**Comment by nicoespeon** [comment]
@MarcZenn the fix is out with v1.8.1: https://github.com/amwmedia/plop/releases/tag/v1.8.1

Let us know how it turns out 😉
